nipype/pipeline/engine/tests/test_engine.py

```diff
@@ -456,6 +456,22 @@ def test_mapnode_iterfield_check():
     with pytest.raises(ValueError): mod1._check_iterfield()
 
 
+@pytest.mark.parametrize("x_inp, f_exp", [
+        (3, [9]), ([2, 3], [4, 9]), ((2,3), [4,9]), (range(3), [0, 1, 4])
+        ])
+def test_mapnode_iterfield_type(x_inp, f_exp):
+    from nipype import MapNode, Function
+    def square_func(x):
+        return x ** 2
```

Comment:
What about make it:

```Python
def double_func(x):
    return x * 2
double = Function(["x"], ["f_x"], double_func)
```

Then you can test that strings are treated as expected, as well. (`'x' * 2 == 'xx'`)